### PR TITLE
Jruby changes

### PR DESCRIPTION
**Pull request by vext01**
These changes are needed for the new version of JRuby that Laurie merged this morning. See commit messages for more details.

Will assign Sarah, as Laurie is super-busy with meetings today.

OK?

### CODE REVIEW

File: krun/vm_defs.py [hunk 1]
```diff
@@ -526,7 +526,7 @@ def __init__(self, vm_path, java_path, env=None):
         JRubyVMDef.__init__(self, vm_path, env=env)
         self.add_env_change(EnvChangeAppend("JAVACMD", java_path))
 
-        self.extra_vm_args += ['-X+T', '-J-server']
+        self.extra_vm_args += ['-J-Djvmci.Compiler=graal', '-X+T']
```
Comment (snim2) [inline krun/vm_defs.py]: Have you tried the JIT test without `-J-Djvmci.Compiler=graal`?

Comment (vext01) [inline krun/vm_defs.py]: Yeah, it isn't strictly needed in my experience, but since both the docs and Chris say I should use it, I did ;)

**Comment by snim2** [comment]
OK, this looks fine to me. Do you need to squash?

**Comment by vext01** [comment]
Nah, I made tidy little commits for once :) Go ahead.

**Pull request merged**